mlrun/db/httpdb.py

```diff
@@ -281,10 +285,14 @@ def api_call(
                     if isinstance(dict_[key], enum.Enum):
                         dict_[key] = dict_[key].value
 
-        # if the method is POST, we need to update the session with the appropriate retry policy
-        if not self.session or method == "POST":
-            retry_on_post = self._is_retry_on_post_allowed(method, path)
-            self.session = self._init_session(retry_on_post)
+        retry_on_post = self._is_retry_on_post_allowed(method, path)
+
+        # PUT retries are allowed unless the path is in NON_RETRIABLE_PATHS
```

Comment:
This comment just tells how `_is_retry_put_allowed` is implemented, it doesn't matter here really.